pkg/controllermanager/controller/add.go

```diff
@@ -31,7 +31,7 @@ import (
 // AddControllersToManager adds all controller-manager controllers to the given manager.
 func AddControllersToManager(mgr manager.Manager, cfg *config.ControllerManagerConfiguration) error {
 	if err := (&bastion.Reconciler{
-		Config: cfg.Controllers.Bastion,
+		Config: *cfg.Controllers.Bastion,
```

Comment:
Would this panic when the bastion controller is disabled?
I#m not entirely sure, but I found a [config option for enabling](https://github.com/gardener/gardener/blob/9044fe3cd5c7c59871bc4de9aca914686f0dc4d3/charts/gardener/controlplane/charts/runtime/templates/controller-manager/configmap-componentconfig.yaml#L34) it. If it was disabled, this would be nil.

Reply:
Setting the bastion controller config is optional in the helm chart, that's correct.
However, the bastion section of controller-manager's component config is defaulted to be non-nil:
https://github.com/gardener/gardener/blob/50f92c5dc35160fe05da9002a79e7ce4a9cf3509/pkg/controllermanager/apis/config/v1alpha1/defaults.go#L31-L40

Hence, the bastion controller is always enabled and these statements will never panic.